src/app/application.cpp

```diff
@@ -144,13 +144,15 @@ Application::Application(const QString &id, int &argc, char **argv)
     QPixmapCache::setCacheLimit(PIXMAP_CACHE_SIZE);
 #endif
 
+    m_portableModeEnabled = QDir(QCoreApplication::applicationDirPath()).exists(DEFAULT_PORTABLE_MODE_PROFILE_DIR);
+
     validateCommandLineParameters();
 
-    const QString profileDir = m_commandLineArgs.portableMode
+    const QString profileDir = m_portableModeEnabled && m_commandLineArgs.profileDir.isEmpty()
         ? QDir(QCoreApplication::applicationDirPath()).absoluteFilePath(DEFAULT_PORTABLE_MODE_PROFILE_DIR)
         : m_commandLineArgs.profileDir;
     Profile::initialize(profileDir, m_commandLineArgs.configurationName,
-                        m_commandLineArgs.relativeFastresumePaths || m_commandLineArgs.portableMode);
+                        m_commandLineArgs.relativeFastresumePaths || m_portableModeEnabled);
```

Comment:
Please help us fix the coding style by adding parentheses to the last parameter.

Reply:
Seems like wrong logic here.
IIRC, if user specifies profile dir via command line it will used even if default profile dir is found. But if user doesn't specify "relative fastresume" param then it doesn't want it to use.
I would change workflow to the following:
1. Process profile related command-line parameters first. If no one is used then
2. Try to find default profile folder.

Reply:
Not sure if I understand what the logic should be with this `relativeFastresumePaths` command line parameter. As I understand `portable` mode should always use `relativeFastresumePaths`, but specifying `--profile=` command line parameter doesn't mean that `relativeFastresumePaths` should be used if it's not specified. Is that correct? If so, then maybe moving check `&& m_commandLineArgs.profileDir.isEmpty()` from https://github.com/Tester798/qBittorrent/blob/e707748f7a56999697e6314ba60efdbc6198289a/src/app/application.cpp#L151 to https://github.com/Tester798/qBittorrent/blob/e707748f7a56999697e6314ba60efdbc6198289a/src/app/application.cpp#L147 would do it?

Reply:
Ok, moved `m_commandLineArgs.profileDir.isEmpty()` check to the top.
Let me know if I should change anything else.